components/gcp-click-to-deploy/src/DeployForm.tsx

```diff
@@ -14,868 +14,1120 @@ import * as React from 'react';
+                    await wait(retryTimeout);
+                }
+                this._appendLine(
+                    `Need to enable these services: ${servicesToEnable.join(', ')}..`);
+            }
+
+            for (const s of servicesToEnable) {
+                this._appendLine('Enabling ' + s);
+                try {
+                    await Gapi.sautil.enableServices(project, saToken, s);
+                } catch (err) {
+                    this._appendLine('Pending on new service account token sync up');
+                }
+            }
+            enableAttempts++;
+        } while (servicesToEnable.length && enableAttempts < 50);
+
+        if (servicesToEnable.length && enableAttempts >= 50) {
+            this.setState({
+                dialogAsCode: false,
+                dialogBody: 'Tried too many times to enable these services: ' +
+                    servicesToEnable.join(', '),
+                dialogTitle: 'Deployment Error',
+            });
+            throw new Error('Deployment Error');
+        }
         this._appendLine(
-          `Need to enable these services: ${servicesToEnable.join(', ')}..`);
-      }
-
-      for (const s of servicesToEnable) {
-        this._appendLine('Enabling ' + s);
-        await Gapi.sautil.enableServices(project, token, s)
-          .catch(e => this._appendLine('Pending on new service account token sync up'));
-      }
-
-      if (this.state.dialogTitle) {
-        return;
-      }
-
-      enableAttempts++;
-    } while (servicesToEnable.length && enableAttempts < 50);
-
-    if (servicesToEnable.length && enableAttempts >= 50) {
-      this.setState({
-        dialogBody: 'Tried too many times to enable these services: ' +
-          servicesToEnable.join(', '),
-        dialogTitle: 'Deployment Error',
-      });
-      return;
+            'All GCP services are available and configured successfully!');
     }
 
-    if (this.state.dialogTitle) {
-      return;
-    }
+    /**
+     * Helper method to facilitate building a v0.5.0 cluster
+     * @param configSpec
+     */
+    private _getV5BuildRequest(configSpec: {defaultApp: string}): string {
+        const {deploymentName, email, project, projectNumber,
+            kfversion, saClientId, saToken} = this.state;
+        let createBody = {
+            AppConfig: configSpec.defaultApp,
+            Apply: true,
+            AutoConfigure: true,
+            Cluster: deploymentName,
+            Email: email,
+            IpName: deploymentName + '-ip',
+            KfVersion: kfversion,
+            Name: deploymentName,
+            Namespace: 'kubeflow',
+            Project: project,
+            ProjectNumber: projectNumber,
+            SAClientId: saClientId,
+            Token: saToken,
+            UseKfctl: this.state.kfctlLib,
+            Zone: this.state.zone,
+        };
+        if (this.state.ingress === IngressType.Iap) {
+         
```

Comment:
Ideally we should pass pasword hash here instead of plain password.
And password type is SecretRef. For example: https://github.com/kubeflow/kubeflow/pull/3830/files#diff-a7d91e7a1b9afdcbb5b17f72957d5863R122

Reply:
Not a problem. I can definitely make the change. To this point, I haven't been able to get a request to work regardless of the auth method but I'll get this in the next change.